atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/InMemoryKeyValueService.java

```diff
@@ -431,7 +431,10 @@ private void putInternal(
 
     @Override
     public CheckAndSetCompatibility getCheckAndSetCompatibility() {
-        return CheckAndSetCompatibility.SUPPORTS_DETAIL_NOT_CONSISTENT_ON_FAILURE;
+        return CheckAndSetCompatibility.supportedBuilder()
+                .supportsDetailOnFailure(true)
+                .consistentOnFailure(true)
```

Comment:
false to match Cassandra for tests?

Reply:
Sure. It's kind of horrible and could be argued that the better way to do that is to have a `TestingInMemoryKeyValueService`, but that's probably too much work.